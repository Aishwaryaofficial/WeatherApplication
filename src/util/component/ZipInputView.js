import React, { useState } from 'react';
import { View,TextInput, Alert ,Button} from 'react-native';
import AppStyle from '../AppStyle';
import AppConstant from '../AppConstant';


const ZipInputView = props => {

    const [enteredText, setEnteredText] = useState('');
    
    const zipInputHandler = (enteredText) => {
        setEnteredText(enteredText);
    }
    const submitHandler = () => {
        const zipCode = enteredText.trim();
        if (zipCode == '') {
            Alert.alert(
                'Zip Code Error!',
                'Zip Code cannot be left blank.'
            );
            return;
        } else if (zipCode.length != 6) {
            Alert.alert(
                'Zip Code Error!',
                'Please enter 6 digit Zip Code to continue.'
            );
            return;
        }
        props.onClick(enteredText);
    }
    return (
        <>
            <View style={AppStyle.zipContainer}>
                <TextInput
                    style={AppStyle.textInput}
                    placeholder={'Enter zipcode '}
                    keyboardType={'number-pad'}
                    returnKeyType={'done'}
                    value={enteredText}
                    onChangeText={zipInputHandler}>
                </TextInput>
                <Button
                    style={AppStyle.submit}
                    title={'submit'}
                    onPress={submitHandler}
                >
                </Button>
            </View>
        </>
    );
};

export default ZipInputView;


