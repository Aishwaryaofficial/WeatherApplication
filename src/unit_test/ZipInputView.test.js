import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import ZipInputView from '../util/component/ZipInputView';

configure({adapter: new Adapter()});
describe('Testing ZipInputView component', () => {
  it('renders as expected', () => {
    const textField = shallow(
      <ZipInputView value='302017' onPress= {jest.fn} />
    );
    expect(textField).toMatchSnapshot();
    textField.setProps({ value: '122003' });
    expect(textField).toMatchSnapshot();


  
  });
});
