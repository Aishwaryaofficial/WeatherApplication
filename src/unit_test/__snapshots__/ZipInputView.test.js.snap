// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ZipInputView component renders as expected 1`] = `
<Fragment>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter zipcode for India"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "backgroundColor": "#e9eef0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 16,
          "height": "80%",
          "marginHorizontal": 15,
          "padding": 5,
          "textAlign": "left",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Button
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#2595ba",
          "borderColor": "#e9eef0",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "tintColor": "#e9eef0",
        }
      }
      title="submit"
    />
  </View>
</Fragment>
`;

exports[`Testing ZipInputView component renders as expected 2`] = `
<Fragment>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter zipcode for India"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "backgroundColor": "#e9eef0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 16,
          "height": "80%",
          "marginHorizontal": 15,
          "padding": 5,
          "textAlign": "left",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Button
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#2595ba",
          "borderColor": "#e9eef0",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "tintColor": "#e9eef0",
        }
      }
      title="submit"
    />
  </View>
</Fragment>
`;
